task/06-conditions-n-loops-tasks.js

```diff
@@ -296,10 +340,31 @@ function getDigitalRoot(num) {
  *   '{[(<{[]}>)]}' = true 
  */
 function isBracketsBalanced(str) {
-    throw new Error('Not implemented');
+
+    let matchingOpeningBracket, ch;
+    let stack = [];
+
+    let openingBrackets = ['[', '{', '(', '<'];
+    let closingBrackets = [']', '}', ')', '>'];
+
+    for (let i = 0; i < str.length; i++) {
+        ch = str[i];
+
+        if (closingBrackets.indexOf(ch) > -1) {
+            matchingOpeningBracket = openingBrackets[closingBrackets.indexOf(ch)];
+            if (stack.length === 0 || (stack.pop() !== matchingOpeningBracket)) {
```

Comment:
`if (!stack.length || (stack.pop() !== matchingOpeningBracket)) {`  this will work